tests/config/pylint_config/test_pylint_config_generate.py

```diff
@@ -33,6 +40,12 @@ def test_format_of_output(
     monkeypatch: MonkeyPatch, capsys: CaptureFixture[str]
 ) -> None:
     """Check that we output the correct format."""
+    # Monkeypatch everything we don't want to check in this test
+    monkeypatch.setattr(
+        "pylint.config._pylint_config.utils.get_and_validate_output_file",
+        lambda: (False, Path()),
+    )
```

Comment:
Shouldn't we separate functions so they are easily testable ? I.e. we provide the result of ``get_and_validate_output_file`` to the function.

Reply:
I'm not sure I understand you. What functions do you want to separate?

Reply:
``generate_interactive_config`` and ``get_and_validate_output_file `` so the former take the result of the latter as argument and it's easier to test it without mocking user outputs.

Reply:
Those are currently separated right? `generate_interactive_config` is the handler for the general `generate --interactive` command, while each subsequent question or set of questions has its own `util` function. So for every set of questions we can simply mock the function and continue.

We could test those `utils` directly instead of using `Run()`? Is that what you mean? I chose not to do this as this makes it easier to test combinations of response: just remove the mock and add the desired answers to `answers`.

Reply:
> We could test those utils directly instead of using Run()? Is that what you mean?

Yes :) But never-mind if it was voluntary !